src/peer.rs

```diff
@@ -90,11 +90,11 @@ pub(crate) fn new_peer<T: HandshakeHandler>(
             active_connections.listeners.clone()
         };
         //HANDSHAKE
-        let peer_id = match handshake_handler.perform_handshake(
+        let peer_id = match handshake_handler.perform_handshake::<M>(
```

Comment:
As `message_handler` is typed above, is it necessary to put the `::<M>` brackets here ?

Reply:
No, I removed it. Thank you :)